### PR TITLE
build path base on BFS through adjancency list (#1669)

### PR DESCRIPTION
**Pull request by nevermore3**
* build path base on BFS through adjancency list

* fix error

* add multi-thread

* fix error

* change filter path

* fix ci test error

* address comment

Co-authored-by: Yee <2520865+yixinglu@users.noreply.github.com>

## What type of PR is this?
- [ ] bug
- [ ] feature
- [x] enhancement

## Checklist:
Tests:
- [ ] Unit test(positive and negative cases)
- [ ] Function test
- [ ] Performance test
- [ ] N/A

Affects:
- [ ] Documentation affected (Please add the label if documentation needs to be modified.)
- [ ] Incompatibility (If it breaks the compatibility, please describe it and add the label.）
- [ ] If it's needed to cherry-pick (If cherry-pick to some branches is required, please label the destination version(s).)
- [ ] Performance impacted: Consumes more CPU/Memory

## Release notes:

Please confirm whether to be reflected in release notes and how to describe:
> ex. Fixed the bug .....

### CODE REVIEW

**Comment by yixinglu** [comment]
@nevermore3 plz fix the tck tests

**Comment by nevermore3** [comment]
> @nevermore3 plz fix the tck tests

done

**Review by yixinglu — APPROVED** [review]

**Review by Aiee — COMMENTED** [review]
Consider use `robin_hood::unordered_flat_` to replace `std::set` and `std::map`

**Review by Aiee — COMMENTED** [review]
Consider use `robin_hood::unordered_flat_` to replace `std::set` and `std::map`

**Review by jievince — APPROVED** [review]

File: src/graph/executor/query/TraverseExecutor.cpp [hunk 1]
```diff
@@ -22,54 +22,61 @@ folly::Future<Status> TraverseExecutor::execute() {
...
+    const auto& metaVidType = *(spaceInfo.spaceDesc.vid_type_ref());
+    auto vidType = SchemaUtil::propTypeToValueType(metaVidType.get_type());
+    for (; iter->valid(); iter->next()) {
+      const auto& vid = src->eval(ctx(iter));
+      if (vid.type() != vidType) {
+        LOG(ERROR) << "Mismatched vid type: " << vid.type() << ", space vid type: " << vidType;
+        continue;
+      }
+      vids_.emplace_back(vid);
+      initVids_.emplace_back(vid);
```
[the remark this answers is not in the document]
Comment (nevermore3) [reply src/graph/executor/query/TraverseExecutor.cpp:74]: according to the design of opencypher, this situation does not report an error

**Pull request merged**